pkg/settings/settings.go

```diff
@@ -66,15 +66,15 @@ func (settings *Settings) AcquirePackageCacheLock() error {
 	if err != nil {
 		return err
 	}
-
+	firstTry := true
 	// if failed to lock the 'package-cache' file, wait until it is unlocked.
 	if !locked {
-		reporter.Report("kpm: waiting for package-cache lock...")
 		for {
 			// try to lock the 'package-cache' file
 			locked, err = settings.PackageCacheLock.TryLock()
-			if err != nil {
-				return err
+			if err != nil && firstTry {
```

Comment:
I personally understand that if there is an error here, there may be an unrecoverable internal error. At this point, it is necessary to exit the loop and report the error instead of trying lock again. The situation of retrying should be that the filelock is locked.

Reply:
fixed.